### PR TITLE
Eliminate more instances of IWorkspaceOptionsProvider

### PR DESCRIPTION
**Pull request by tmat**
All options are now passed explicitly to the APIs that need them, except for default values for editorconfig-based formatting options. These will be eliminated later on.

Requires Roslyn changes: https://github.com/dotnet/roslyn/pull/59469

### CODE REVIEW

**Comment by tmat** [comment]
@JoeRobich @[ryzngard](https://github.com/ryzngard)

File: src/OmniSharp.Roslyn.CSharp/Services/Completion/CompletionService.cs
Comment (333fred) [inline src/OmniSharp.Roslyn.CSharp/Services/Completion/CompletionService.cs:122]: This change isn't correct, and is the source of the test failures. We need to only mark this as true when the cache is complete and returning full info, which is what the `expandedItemsAvailable` variable was marking. Seems like with @genlu's import change, this may need to be reworked.

Comment (genlu) [reply src/OmniSharp.Roslyn.CSharp/Services/Completion/CompletionService.cs:122]: I will add a flag for partial results from import completion providers

Comment (genlu) [reply src/OmniSharp.Roslyn.CSharp/Services/Completion/CompletionService.cs:122]: Fix PR: https://github.com/dotnet/roslyn/pull/59958
Once this change is merged, O# need to set the ForceExpandedCompletionIndexCreation option to true before calling CompletionService to make test behavior deterministic

**Comment by JoeRobich** [comment]
@333fred @tmat I updated with a build of Roslyn that includes Gen's change. Feedback welcome on https://github.com/OmniSharp/omnisharp-roslyn/pull/2343/commits/43ca31876ec9266f83854b29d4a14245f0e5f3bc

File: tests/OmniSharp.Roslyn.CSharp.Tests/CompletionFacts.cs
Comment (333fred) [inline tests/OmniSharp.Roslyn.CSharp.Tests/CompletionFacts.cs:178]: We could likely avoid this entire loop if we just set the option to true.

Comment (JoeRobich) [reply tests/OmniSharp.Roslyn.CSharp.Tests/CompletionFacts.cs]: I ended up doing that for `FindCompletionsWithImportedAsync()`. Will let this test remain our test that the index will eventually be created.

**Review by JoeRobich — APPROVED** [review]

File: src/OmniSharp.Roslyn.CSharp/Services/Completion/CompletionService.cs
Comment (333fred) [reply src/OmniSharp.Roslyn.CSharp/Services/Completion/CompletionService.cs:129]: This condition is still incorrect. We need to know if we got partial results back from the `GetCompletionsAsync`, or `IsIncomplete` will not be correct.

Comment (genlu) [reply src/OmniSharp.Roslyn.CSharp/Services/Completion/CompletionService.cs:129]: I was just discussing this with @JoeRobich offline. You are right that the `IsIncomplete` flag will not be correct, but you couldn't decide if the result is complete with the removed "expandedItemsAvailable" flag either. That flag only means import completion would trigger in current context, it doesn't mean the result the provider returns is complete or any results would be returned at all. 

The behavior for unimported types was "all or nothing" before (depends on cache state), which might gives you an impression that the existence of unimported type is an indication of complete results, but it's just an implementation detail and was changed in [this PR](https://github.com/dotnet/roslyn/pull/59125) to return partial results, which is consistent with how unimported extension methods works.

If you'd like to support `IsIncomplete` flag, maybe take a look at the approach used in [this PR](https://github.com/dotnet/roslyn/pull/59045)

**Review by genlu — APPROVED** [review]
Completion part LGTM

File: src/OmniSharp.Roslyn.CSharp/Services/Completion/CompletionService.cs
Comment (333fred) [reply src/OmniSharp.Roslyn.CSharp/Services/Completion/CompletionService.cs:129]: Ok. I'll try to look at that at a later date, we don't need to block this PR on it. At worst, the first few completion sessions will be inconsistent, but following sessions will be fine.

**Review by 333fred — APPROVED** [review]

**Pull request merged**
